src/node/stage/approved.rs

```diff
@@ -586,7 +544,7 @@ impl Approved {
         let bounce_msg = bounce_msg.to_bytes();
 
         if let Some(sender) = sender {
-            self.comm.send_message_to_target(&sender, bounce_msg).await
```

Comment:
> any special handling required?

Yes, it needs to vote `Offline` for lost peers, because they are now detected inside `Comm::send_message_to_targets`.

Reply:
but seems the most functions are duplicated.

anyway, it can be addressed later on to see if can have a common `Comm` to be used by all.